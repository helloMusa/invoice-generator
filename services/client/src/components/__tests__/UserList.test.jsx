import React from "react";
import { render, cleanup } from "@testing-library/react";

import UsersList from "../UsersList";

afterEach(cleanup);

const users = [
  {
    email: "musa6ali@gmail.com",
    id: 1,
    username: "musa",
  },
  {
    email: "musa@mali.org",
    id: 2,
    username: "musaali",
  },
];

it("renders a username", () => {
  const { getByText } = render(<UsersList users={users} />);
  expect(getByText("musa")).toHaveClass("username");
  expect(getByText("musaali")).toHaveClass("username");
});

it("renders", () => {
  const { asFragment } = render(<UsersList users={users} />);
  expect(asFragment()).toMatchSnapshot();
});
