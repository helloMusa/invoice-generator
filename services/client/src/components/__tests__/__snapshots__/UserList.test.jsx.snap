// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <p
      class="box title is-4 username"
    >
       musa
    </p>
    <p
      class="box title is-4 username"
    >
       musaali
    </p>
  </div>
</DocumentFragment>
`;
